src/qr-generator.coffee

```diff
@@ -31,7 +31,11 @@ module.exports = (robot) ->
 
     urlObj = makeUrlObj data
     hackUrlObj = adapterHack urlObj, robot.adapterName
-    msg.send url.format hackUrlObj
+    mapUrl = url.format hackUrlObj
+    if mdOut
```

Comment:
I would introduce a message formatter function or object. Which based on a formatter defined through environment var, formats a message and returns the formatted message. 
Keeps the 'main' logic nice and clean.